### PR TITLE
Log errors during store-gateway and ingester chunks streaming to logs, not just trace spans

### PR DESCRIPTION
**Pull request by charleskorn**
#### What this PR does

Split from https://github.com/grafana/mimir/pull/6194 (see https://github.com/grafana/mimir/pull/6194#discussion_r1345389927)

#### Which issue(s) this PR fixes or relates to

(none)

#### Checklist

- [n/a] Tests updated
- [n/a] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/ingester/client/streaming.go [hunk 1]
```diff
@@ -90,6 +90,7 @@ func (s *SeriesChunksStreamReader) StartBuffering() {
 		onError := func(err error) {
 			s.errorChan <- err
 			log.Error(err)
```
Comment (dimitarvdimitrov) [inline pkg/ingester/client/streaming.go:92]: can we replace this with `ext.Error.Set(log.Span, true)`

**Review by dimitarvdimitrov — COMMENTED** [review]
i left a suggestion; another note is to maybe add a comment explaining that these aren't double logs, but each has a slightly distinct purpose

File: pkg/ingester/client/streaming.go (hunk 1 above)
Comment (charleskorn) [reply pkg/ingester/client/streaming.go]: Yep, that's a nicer solution, thanks.

**Comment by charleskorn** [comment]
> So my question is: what if improve `SpanLogger.Log()` (specifically `SpanLogger.spanLog()`) to auto-detect the `"err"` field and, if valued, call `ext.Error.Set(s.Span, true)` from there? I guess a downside of this approach is that it may be considered a breaking change.

Yeah, I think this would be a breaking change, and possibly undesirable in some situations.

For example, imagine a situation where we're retrying a network call and log something like `level.Warn(log).Log("msg", "request failed, retrying", "attempt", attemptCount, "err", err)` after an early failure - we wouldn't want to set `Error` to true in this case.

Perhaps we could do something like looking for a particular severity and the presence of an `err` field, but then there might be other situations where that's still not quite right.

**Comment by charleskorn** [comment]
I've manually backported this in #6327.

**Pull request merged**